request-manager/graph/grapher.go

```diff
@@ -287,30 +310,29 @@ func buildSeqGraph(seqSpec *spec.Sequence, idgen id.Generator) (seqGraph *Graph,
 		}
 	}
 
-	// Append sink node to ordering
 	seqGraph.Order = append(seqGraph.Order, seqGraph.Sink)
 
 	// Make sure we haven't created a deformed graph
 	// If we do, it's a bug in the code, not a problem with the specs
 	if !seqGraph.IsValidGraph() {
-		return nil, nil, fmt.Errorf("malformed graph created")
+		return nil, nil, fmt.Errorf("malformed sequence graph created")
```

Comment:
We need to improve this by making `IsValidGraph()` return `error` that describes the problem. So, `return nil, nil, fmt.Errorf("sequence %s is not a valid directed acyclic graph: %s", err)`